src/core/send.c

```diff
@@ -529,12 +528,29 @@ QuicSendWriteFrames(
             IsApplicationClose = FALSE;
         }
 
+        QUIC_VAR_INT CloseErrorCode = Connection->CloseErrorCode;
+        char* CloseReasonPhrase = Connection->CloseReasonPhrase;
+
+        if (IsApplicationClose && ! Is1RttEncryptionLevel) {
+            // 
+            // A CONNECTION_CLOSE of type 0x1d MUST be replaced by a
+            // CONNECTION_CLOSE of type 0x1c when sending the frame in Initial or
+            // Handshake packets. Otherwise, information about the application
+            // state might be revealed. Endpoints MUST clear the value of the
+            // Reason Phrase field and SHOULD use the APPLICATION_ERROR code when
+            // converting to a CONNECTION_CLOSE of type 0x1c.
+            //
+            CloseErrorCode = QUIC_ERROR_APPLICATION_ERROR;
```

Comment:
Why aren't we using the `QUIC_ERROR_CRYPTO_USER_CANCELED` error? I thought that's what is supposed to be used.

Reply:
See [RFC](https://datatracker.ietf.org/doc/html/rfc9000#section-10.2.3-3)

> Sending a CONNECTION_CLOSE of type 0x1d in an Initial or Handshake packet could expose application state or be used to alter application state. A CONNECTION_CLOSE of type 0x1d MUST be replaced by a CONNECTION_CLOSE of type 0x1c when sending the frame in Initial or Handshake packets. Otherwise, information about the application state might be revealed. Endpoints MUST clear the value of the Reason Phrase field and SHOULD use the APPLICATION_ERROR code when converting to a CONNECTION_CLOSE of type 0x1c.

Follow up question: should we introduce also equivalent QUIC_STATUS code?

Reply:
This must have changed after we wrote the USER_CANCELED code, and then we just never updated to match.
> Follow up question: should we introduce also equivalent QUIC_STATUS code?
No. Let's leave it as is for now.